lib/widgets/profile.dart

```diff
@@ -33,6 +35,36 @@ class ProfilePage extends StatelessWidget {
       page: ProfilePage(userId: userId));
   }
 
+  /// The given user's real email address, if known, for displaying in the UI.
+  ///
+  /// Returns null if [selfUser] isn't able to see [user]'s real email address.
+  ///
+  /// **Note:** Starting from Zulip version 7.0 (FL 163), there's a change in
+  /// the API about how [selfUser] can access [user]'s real email address.
+  /// Search for "delivery_email" in https://zulip.com/api/register-queue.
+  String? _getDisplayEmailFor(User user, {required PerAccountStore store}) {
+    if (store.account.zulipFeatureLevel >= 163) {
```

Comment:
Let's include a `TODO(server-7)` for removing the <163 case. That way, when we decide to bump our minimum-supported threshold to 7, this spot will naturally show up in a quick search for backwards-compatibility code that can be removed at that time.